Milvus.Client/ArrayFieldData.cs

```diff
@@ -0,0 +1,187 @@
+﻿namespace Milvus.Client;
+
+/// <summary>
+/// Binary Field
+/// </summary>
+public sealed class ArrayFieldData<TElementData> : FieldData<IReadOnlyList<TElementData>>
+{
+    /// <summary>
+    /// Construct an array field
+    /// </summary>
+    /// <param name="fieldName"></param>
+    /// <param name="data"></param>
+    /// <param name="isDynamic"></param>
+    public ArrayFieldData(string fieldName, IEnumerable<IEnumerable<TElementData>> data, bool isDynamic)
+        : base(fieldName, data.Select(x => x.ToArray()).ToArray(), MilvusDataType.Array, isDynamic)
+    {
+        ElementType = EnsureDataType<TElementData>();
+    }
+
+    /// <summary>
+    /// Array element type
+    /// </summary>
+    public MilvusDataType ElementType { get; }
+
+    /// <inheritdoc />
+    internal override Grpc.FieldData ToGrpcFieldData()
+    {
+        Check();
+
+        Grpc.FieldData fieldData = new()
+        {
+            Type = Grpc.DataType.Array,
+            IsDynamic = IsDynamic
+        };
+
+        if (FieldName is not null)
+        {
+            fieldData.FieldName = FieldName;
+        }
+
+        var arrayArray = new ArrayArray
+        {
+            ElementType = (DataType) ElementType,
+        };
+
+        fieldData.Scalars = new ScalarField
+        {
+            ArrayData = arrayArray
+        };
+
+        foreach (var array in Data)
+        {
+            switch (ElementType)
+            {
+                case MilvusDataType.Bool:
+                    Grpc.BoolArray boolData = new();
+                    boolData.Data.AddRange(array as IEnumerable<bool>);
+                    arrayArray.Data.Add(new ScalarField {BoolData = boolData});
```

Comment:
nit (below as well):

```suggestion
                    arrayArray.Data.Add(new ScalarField { BoolData = boolData });
```

Reply:
Changed code style